waiter/src/waiter/scheduler.clj

```diff
@@ -370,6 +370,16 @@
     (if-not service
       service->service-instances'
       (let [{:strs [health-check-url]} (service-id->service-description-fn (:id service))
+            update-flags (fn [instance error] 
+                           (update-in instance [:flags] 
```

Comment:
Is it possible for an instance to have conflicting flags? If so, is there a sanity check somewhere?

Reply:
For unhealthy instances, there are no conflicting flags.

Reply:
Can we have a separate function that processes unhealthy instances?

Reply:
The instance is closed by the function, so it can't be removed without a more significant refactoring.

Reply:
Does this not work?
```
(defn foo-bar-on-unhealthy-instance [instance status error]
  (-> instance
       (assoc :healthy? false)
       (assoc :health-check-status status)
       (update-in instance [:flags] 
                                      (fn [flags]
                                        (cond-> flags
                                          (not= error :connect-exception)
                                          (conj :has-connected)

                                          (and (not= error :connect-exception) (not= error :timeout-exception))
                                          (conj :has-responded)))))
```